crates/oxc_prettier/src/format/call_expression.rs

```diff
@@ -88,31 +92,140 @@ fn print_call_expression_arguments<'a>(
         return Doc::Array(parts);
     }
 
-    let mut parts_inner = p.vec();
+    #[allow(clippy::cast_sign_loss)]
+    let get_printed_arguments = |p: &mut Prettier<'a>, skip_index: isize| {
+        let mut printed_arguments = p.vec();
+        let mut len = arguments.len();
+        let arguments: Box<dyn Iterator<Item = (usize, &Argument)>> = match skip_index {
```

Comment:
In some cases, we don't need to format all arguments. This is because some arguments require special handling. Therefore, "skip_index" skips the unneeded arguments.

The `get_printed_arguments` function's is meant to address the ownership rules in `Rust`.

Reply:
> In some cases, we don't need to format all arguments. This is because some arguments require special handling. Therefore, `skip_index` skips the unneeded arguments.
> 
> The `get_printed_arguments` function is meant to address the ownership rules in `Rust`.

Reply:
In some cases, we don't need to format all arguments. This is because some arguments require special handling. Therefore, `skip_index` skips the unneeded arguments.
The `get_printed_arguments` function is meant to address the ownership rules in `Rust`.

Reply:
Would you mind adding this explain as a comment?